### ISSUE TITLE
bootup error

### ISSUE THREAD

**Issue by johnfg2610** [simon816/SpongeBootstrap#8]
running mc 1.11.2 
```
01.08 11:14:53 [Server] main/ERROR [FML]: An error occurred trying to configure the minecraft home at /. for Forge Mod Loader
01.08 11:14:53 [Server] INFO java.lang.NoClassDefFoundError: org/objectweb/asm/ClassVisitor
01.08 11:14:53 [Server] INFO at net.minecraftforge.fml.relauncher.CoreModManager.discoverCoreMods(CoreModManager.java:349) ~[forge-1.11.2-13.20.1.2386-universal.jar:?]
01.08 11:14:53 [Server] INFO at net.minecraftforge.fml.relauncher.CoreModManager.handleLaunch(CoreModManager.java:255) ~[forge-1.11.2-13.20.1.2386-universal.jar:?]
01.08 11:14:53 [Server] INFO at net.minecraftforge.fml.relauncher.FMLLaunchHandler.setupHome(FMLLaunchHandler.java:100) [forge-1.11.2-13.20.1.2386-universal.jar:?]
01.08 11:14:53 [Server] INFO at net.minecraftforge.fml.relauncher.FMLLaunchHandler.setupServer(FMLLaunchHandler.java:85) [forge-1.11.2-13.20.1.2386-universal.jar:?]
01.08 11:14:53 [Server] INFO at net.minecraftforge.fml.relauncher.FMLLaunchHandler.configureForServerLaunch(FMLLaunchHandler.java:47) [forge-1.11.2-13.20.1.2386-universal.jar:?]
01.08 11:14:53 [Server] INFO at net.minecraftforge.fml.common.launcher.FMLServerTweaker.injectIntoClassLoader(FMLServerTweaker.java:62) [forge-1.11.2-13.20.1.2386-universal.jar:?]
01.08 11:14:53 [Server] INFO at net.minecraft.launchwrapper.Launch.launch(Launch.java:115) [launchwrapper-1.12.jar:?]
01.08 11:14:53 [Server] INFO at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
01.08 11:14:53 [Server] INFO at com.simon816.sponge.bootstrap.Bootstrap.load(Bootstrap.java:152) [spongebootstrap.jar:?]
01.08 11:14:53 [Server] INFO at com.simon816.sponge.bootstrap.Bootstrap.main(Bootstrap.java:56) [spongebootstrap.jar:?]
01.08 11:14:53 [Server] INFO Caused by: java.lang.ClassNotFoundException: org.objectweb.asm.ClassVisitor
01.08 11:14:53 [Server] INFO at java.net.URLClassLoader.findClass(URLClassLoader.java:381) ~[?:1.8.0_111]
01.08 11:14:53 [Server] INFO at java.lang.ClassLoader.loadClass(ClassLoader.java:424) ~[?:1.8.0_111]
01.08 11:14:53 [Server] INFO at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331) ~[?:1.8.0_111]
01.08 11:14:53 [Server] INFO at java.lang.ClassLoader.loadClass(ClassLoader.java:357) ~[?:1.8.0_111]
01.08 11:14:53 [Server] INFO ... 10 more
01.08 11:14:53 [Server] INFO Exception in thread "main" [11:14:53] [main/INFO] [STDERR]: [java.lang.ThreadGroup:uncaughtException:1052]: java.lang.NoClassDefFoundError: org/objectweb/asm/ClassVisitor
01.08 11:14:53 [Server] main/INFO [STDERR]: [java.lang.ThreadGroup:uncaughtException:1052]: 	at net.minecraftforge.fml.relauncher.CoreModManager.discoverCoreMods(CoreModManager.java:349)
01.08 11:14:53 [Server] main/INFO [STDERR]: [java.lang.ThreadGroup:uncaughtException:1052]: 	at net.minecraftforge.fml.relauncher.CoreModManager.handleLaunch(CoreModManager.java:255)
01.08 11:14:53 [Server] main/INFO [STDERR]: [java.lang.ThreadGroup:uncaughtException:1052]: 	at net.minecraftforge.fml.relauncher.FMLLaunchHandler.setupHome(FMLLaunchHandler.java:100)
01.08 11:14:53 [Server] main/INFO [STDERR]: [java.lang.ThreadGroup:uncaughtException:1052]: 	at net.minecraftforge.fml.relauncher.FMLLaunchHandler.setupServer(FMLLaunchHandler.java:85)
01.08 11:14:53 [Server] main/INFO [STDERR]: [java.lang.ThreadGroup:uncaughtException:1052]: 	at net.minecraftforge.fml.relauncher.FMLLaunchHandler.configureForServerLaunch(FMLLaunchHandler.java:47)
01.08 11:14:53 [Server] main/INFO [STDERR]: [java.lang.ThreadGroup:uncaughtException:1052]: 	at net.minecraftforge.fml.common.launcher.FMLServerTweaker.injectIntoClassLoader(FMLServerTweaker.java:62)
```

**Comment by johnfg2610** [comment]
also getting on 1.12

```
01.08 12:15:52 [Server] INFO java.lang.NoClassDefFoundError: joptsimple/OptionSpec
01.08 12:15:52 [Server] INFO at java.lang.Class.forName0(Native Method)
01.08 12:15:52 [Server] INFO at java.lang.Class.forName(Class.java:264)
01.08 12:15:52 [Server] INFO at com.simon816.sponge.bootstrap.Bootstrap.main(Bootstrap.java:44)
01.08 12:15:52 [Server] INFO Caused by: java.lang.ClassNotFoundException: joptsimple.OptionSpec
01.08 12:15:52 [Server] INFO at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
01.08 12:15:52 [Server] INFO at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
01.08 12:15:52 [Server] INFO at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
01.08 12:15:52 [Server] INFO at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
```

**Comment by simon816** [comment]
Can you verify that forge was installed correctly before adding the bootstrap. Test whether forge loads without sponge or rerun the forge installer

**Comment by johnfg2610** [comment]
forge was installed correctly and i redid the libs after seemed to fix the first issue but not the second

**Comment by simon816** [comment]
I don't think all the files are set-up correctly for your 1.12 server. There should be a file `libraries/net/sf/jopt-simple/jopt-simple/4.6/jopt-simple-4.6.jar`. Does forge launch successfully without bootstrap?

**Comment by johnfg2610** [comment]
Yup seems to be fine without bootstrap

**Comment by simon816** [comment]
Could you pastebin/gist the full forge log file found in `logs/fml-server-latest.log` as well as the minecraft log `logs/latest.log`?

I assume you're using the latest bootstrap (v0.4.0), just make sure it is the latest.

**Comment by simon816** [comment]
Should be fixed now with version 0.5.0
